airflow/providers/google/cloud/hooks/dataflow.py

```diff
@@ -319,48 +319,50 @@ def __init__(
         on_new_job_id_callback: Optional[Callable[[str], None]] = None
     ) -> None:
         super().__init__()
-        self.log.info("Running command: %s", ' '.join(cmd))
+        self.log.info("Running command: %s", ' '.join(shlex.quote(c) for c in cmd))
         self.on_new_job_id_callback = on_new_job_id_callback
+        self.job_id: Optional[str] = None
         self._proc = subprocess.Popen(
             cmd,
             shell=False,
             stdout=subprocess.PIPE,
             stderr=subprocess.PIPE,
             close_fds=True)
 
-    def _read_line_by_fd(self, fd):
-        if fd == self._proc.stderr.fileno():
-            line = self._proc.stderr.readline().decode()
-            if line:
-                self.log.warning(line[:-1])
-            return line
-
-        if fd == self._proc.stdout.fileno():
-            line = self._proc.stdout.readline().decode()
-            if line:
-                self.log.info(line[:-1])
-            return line
+    def process_fb(self, fd):
```

Comment:
```suggestion
    def process_fd(self, fd):
```
Typo? Btw What is the `fd`?

Reply:
File descriptior

> A file descriptor is either a socket or file object, or a small integer
    gotten from a fileno() method call on one of those.

Reply:
I upgrade the PR.